pkg/kubefed2/unjoin.go

```diff
@@ -381,33 +380,32 @@ func deleteServiceAccount(clusterClientset client.Interface, saName,
 // deleteClusterRoleAndBinding deletes an RBAC cluster role and binding that
 // allows the service account identified by saName to access all resources in
 // all namespaces in the cluster associated with clusterClientset.
-func deleteClusterRoleAndBinding(clusterClientset client.Interface, saName, namespace,
-	unjoiningClusterName string, dryRun bool) error {
-
+func deleteClusterRoleAndBinding(clusterClientset client.Interface, saName, namespace string, dryRun bool) error {
 	if dryRun {
 		return nil
 	}
 
 	roleName := util.RoleName(saName)
+	healthzRoleName := util.HealthzRoleName(saName)
 
-	// Attempt to delete both federation and cluster role and role
-	// bindings and ignore if they are missing.
+	// Attempt to delete all role and role bindings created by join
+	// and ignore if they are missing.
 
-	err := clusterClientset.RbacV1().ClusterRoleBindings().Delete(roleName, &metav1.DeleteOptions{})
-	if err != nil && !errors.IsNotFound(err) {
-		glog.V(2).Infof("Could not delete cluster role binding for service account: %s in unjoining cluster: %v",
-			saName, err)
-		return err
-	}
+	for _, name := range []string{roleName, healthzRoleName} {
+		err := clusterClientset.RbacV1().ClusterRoleBindings().Delete(name, &metav1.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			glog.V(2).Infof("Could not delete cluster role binding %q in unjoining cluster: %v", name, err)
+			return err
```

Comment:
Nit: Would it rather make sense to continue and try delete/remove remaining roles/bindings if an error occurs on one. I know this is a tricky bit in unjoin flow, this however makes more sense to me.

We can have a separate discussion on this outside this particular PR. I always felt that unjoin should ideally try and delete/remove all the resources sequentially, only logging failures, and always finishing the sequence, which does not seem to be the case now. If that is the case, unjoin command can be triggered again in case a failure is seen (NotFound is obviously not a failure).

Reply:
I agree that unjoin should be resilient to the case of missing resources.  It wasn't my intent to improve the error handling here (just simplify role handling as you requested in a previous PR), but it is intended to be done as per #188.

Reply:
Ack.